schema/RequestSignInLinkMutation.re

```diff
@@ -18,7 +21,50 @@ let requestSignInLink =
       "requestSignInLink",
       ~typ=non_null(GraphqlTypes.errorPayloadType),
       ~args=Arg.[arg("input", ~typ=non_null(requestSignInLinkInput))],
-      ~resolve=(_, (), input) =>
-      Lwt.return(Ok(None))
+      ~resolve=(context: Context.t, (), email) => {
+        let%lwt user = User.getByEmail(context.connection, email);
+        switch (user) {
+        | Some(user) =>
+          let%lwt (response, _) =
+            Sendgrid.sendEmail(
+              ~to_=email,
+              ~subject="Summhub - Login Access Link",
+              ~content=
+                "<a href=\"https://summhub.com/sign-in/\""
+                ++ Auth.encodeToken(user.id)
+                ++ " > Click here to sign in </a>",
+            );
+          /* Sendgrid uses the code 202 to indicate success otherwise is fail https://sendgrid.com/docs/API_Reference/api_v3.html */
+          switch (response.status) {
+          | `Code(202) => Lwt.return(Ok(None))
+          | _ => Lwt.return(Ok(Some(Errors.errorOnEmailSend)))
+          };
+          Lwt.return(Ok(None));
+        | None =>
+          let%lwt result = User.createUser(context.connection, email);
+          switch (result) {
+          | None => Lwt.return(Ok(Some(Errors.somethingWentWrong)))
+          | Some(_) =>
```

Comment:
we should change the insert method to return the result, https://github.com/Astrocoders/lion-carne-server/blob/develop/src/Helpers/entityFactory.js#L31